mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -343,6 +342,30 @@ private Optional<Map<TaskExecutorID, TaskExecutorState>> findBestFitFor(TaskExec
                     this.taskExecutorStateMap::get)));
     }
 
+    /**
+     * Verifies if all scheduling attributes constraints are satisfied.
+     *
+     * For each entry in 'schedulingAttributes':
+     * - Fetch the corresponding attribute value from Task Executor scheduling attributes, if not present, default to the value from the current entry.
+     * - Fetch the corresponding attribute value from Schedule Request scheduling attributes, if not present, default to the value from the current entry.
+     * - Checks if these two values are equal ignoring case. If any pair is not equal, the function returns false.
+     *
+     * Hence, the function ensures that the TaskExecutor scheduling attributes match or satisfy the constraints required. If either the TaskExecutor registration or the scheduling request lacks an attribute, it uses the provided defaults.
+     *
+     * @param constraints The schedule request constraints to be satisfied.
+     * @param teAssignmentAttributes The scheduling attributes of a Task Executor that needs to satisfy scheduling constraints.
+     *
+     * @return true if all allocation constraints are satisfied, false otherwise.
+     */
+    public boolean areSchedulingAttributeConstraintsSatisfied(SchedulingConstraints constraints, Map<String, String> teAssignmentAttributes) {
+        return schedulingAttributes.entrySet()
```

Comment:
Why do we need to define this as part of the control plane? The job definition already provides the scheduling constraints, including the hard constraints. All we need to do is match the TE attributes against these hard constraints. So, why do we need to duplicate this information in the control plane?